Applications/DataExplorer/NetCdfDialog/NetCdfConfigureDialog.cpp

```diff
@@ -216,81 +200,53 @@ void NetCdfConfigureDialog::setDimensionSelect()
-            size = sizeOfDim;
-            double *arrayOfDimStart = new double[1]; //[1] = {0};
-            long edgeOfArray = 1; //[1] = {1};
-            long edgeOrigin[1] = {0};
-            tmpVarOfDim->set_cur(edgeOrigin);
-            tmpVarOfDim->get(arrayOfDimStart,edgeOfArray);
-            firstValue = arrayOfDimStart[0];
-            double arrayOfDimEnd[1] = {0};
-            edgeOrigin[0] = sizeOfDim - 1;
-            tmpVarOfDim->set_cur(edgeOrigin);
-            tmpVarOfDim->get(arrayOfDimEnd,edgeOfArray);
-            lastValue = arrayOfDimEnd[0];
-        }
-    }else{
-        size = 0;
-        firstValue = 0;
-        lastValue = 0;
+        size = tmpVarOfDim.getDim(0).getSize();
+        std::vector<std::size_t> start_val{{0}};
+        std::vector<std::size_t> length{{1}};
+        tmpVarOfDim.getVar(start_val, length, &firstValue);
+        start_val[0] = size-1;
+        tmpVarOfDim.getVar(start_val, length, &lastValue);
     }
 }
 
-void NetCdfConfigureDialog::getDaysTime(double minSince, QTime &time, int &days)
-{
-    auto tmpMin = (long)minSince;
-    long minuits = tmpMin % 60;
-    long tmpHours = tmpMin / 60;
-    long hours = tmpHours % 24;
-    days = (int)(tmpHours / 24);
-    time.setHMS(hours, minuits, 0);
-}
-
-long NetCdfConfigureDialog::convertDateToMinutes(QDateTime initialDateTime, QDate selectedDate, QTime selectedTime)
+int NetCdfConfigureDialog::getTimeStep() const
 {
-    long tmpInitialToSelectedDate = static_cast<long>(selectedDate.daysTo(initialDateTime.date()));
-    long selectedDays = - tmpInitialToSelectedDate * 24 * 60;
-    long selectedMinutes = (selectedTime.hour() * 60) + selectedTime.minute() + selectedDays;
-    return selectedMinutes;
-}
-
-int NetCdfConfigureDialog::getTimeStep()
-{
-    NcVar* timeVar = _currentFile->get_var(comboBoxDim2->currentIndex());
-
-    double const datesToMinutes = convertDateToMinutes(_currentInitialDateTime,dateTimeEditDim3->date(),dateTimeEditDim3->time());
-
-    double timeArray[1] = {datesToMinutes};
-    double currentTime = timeVar->get_index(timeArray);
-    if (currentTime < 0) currentTime=0; //if the value isn't found in the array, set it to 0 as default...
-    return static_cast<int>(currentTime);
+    return dateTimeEditDim3->value();
 }
 
 int NetCdfConfigureDialog::getDim4() const
 {
-    NcVar* dim3Var = _currentFile->get_var(comboBoxDim4->currentIndex());
-    int timeArray[1] = {spinBoxDim4->value()};
-    int currentValueDim3 = dim3Var->get_index(timeArray);
-    if (currentValueDim3 < 0) currentValueDim3=0; //if the value isn't found in the array, set it to 0 as default...
-    return currentValueDim3;
+    NcVar const& dim3Var =
+        _currentFile.getVar(comboBoxDim4->currentText().toStdString());
+    std::vector<std::size_t> start{{static_cast<std::size_t>(spinBoxDim4->value())}};
+    std::vector<std::size_t> length{{1}};
+    int value(0);
+    dim3Var.getVar(start, length, &value);
```

Comment:
Can you try this, please?
```suggestion
    dim3Var.getVar(start, {1}, &value);
```
If it works, same above around 220...

Reply:
It works :)  
:heavy_check_mark: